examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -88,27 +92,52 @@ namespace {
 // This file is being used by platforms other than Linux, so we need this check to disable related features since we only
 // implemented them on linux.
 #if CHIP_DEVICE_LAYER_TARGET_LINUX
+constexpr EndpointId kNetworkCommissioningEndpointForThread = 0;
+constexpr EndpointId kNetworkCommissioningEndpointForWiFi   = 1;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
 NetworkCommissioning::LinuxThreadDriver sLinuxThreadDriver;
-Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(0 /* Endpoint Id */, &sLinuxThreadDriver);
+Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(kNetworkCommissioningEndpointForThread,
+                                                                             &sLinuxThreadDriver);
 #endif
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
 NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
-Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(1 /* Endpoint Id */, &sLinuxWiFiDriver);
-#endif
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(kNetworkCommissioningEndpointForWiFi, &sLinuxWiFiDriver);
 #endif
+#endif // CHIP_DEVICE_LAYER_TARGET_LINUX
 } // namespace
 
 void ApplicationInit()
 {
 #if CHIP_DEVICE_LAYER_TARGET_LINUX && defined(ZCL_USING_LEVEL_CONTROL_CLUSTER_SERVER)
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-    sThreadNetworkCommissioningInstance.Init();
+    if (LinuxDeviceOptions::GetInstance().mThread)
+    {
+        sThreadNetworkCommissioningInstance.Init();
+    }
+    else
 #endif
+    {
+        uint32_t nullFeatureMapValue = 0;
+        emberAfWriteAttribute(kNetworkCommissioningEndpointForThread, Clusters::NetworkCommissioning::Id,
```

Comment:
Please use Accessors functions instead of type-unsafe emberAfWriteAttribute, if we do end up writing attributes.